x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/delete_custom_dashboard.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createRouteValidationFunction } from '@kbn/io-ts-utils';
+import { InfraDeleteCustomDashboardsRequestParamsRT } from '../../../common/http_api/custom_dashboards_api';
+import { KibanaFramework } from '../../lib/adapters/framework/kibana_framework_adapter';
+import { handleRouteErrors } from '../../utils/handle_route_errors';
+import { checkCustomDashboardsEnabled } from './lib/check_custom_dashboards_enabled';
+import { deleteCustomDashboard } from './lib/delete_custom_dashboard';
+import { findCustomDashboardByDashboardId } from './lib/find_custom_dashboard';
+
+export function initDeleteCustomDashboardRoute(framework: KibanaFramework) {
+  const validateParams = createRouteValidationFunction(InfraDeleteCustomDashboardsRequestParamsRT);
+
+  framework.registerRoute(
+    {
+      method: 'delete',
+      path: '/api/infra/{assetType}/custom-dashboards/{dashboardSavedObjectId}',
```

Comment:
Does `dashboardSavedObjectId` refer to the dashboard id that's linked in a custom saved object with its own id, or `dashboardSavedObjectId` is the id of the custom saved object?

If it's the former, it's probably better to change it to receive the id of the saved object instead. In REST, usually when we have a path parameter that is an id, it's usually the id of the resource itself, ie: the id of the saved object we're deleting in this case

Reply:
I believe it was renamed to dashboardSavedObjectId to illustrate that it's the dashboards saved object id.  We are not setting specific ids so they'll default to something like "2b75ef40-14ca-4f20-a9cc-6a3ed8c436d9".

Reply:
@crespocarlos Thank you for the suggestion! Yes, `dashboardSavedObjectId` refers to the dashboard id (of the linked dashboard). I agree with you in general - here I was thinking that we want to use the `dashboardSavedObjectId` because when we use this endpoint the UI doesn't need to know about the SO id and we can have only the `dashboardSavedObjectId` in use. I saw that in APM we also use the SO id so I will do something similar, it makes sense to avoid doing the extra query.